presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -29,46 +39,112 @@
 import static com.facebook.presto.sql.planner.plan.JoinNode.Type.LEFT;
 import static com.facebook.presto.sql.planner.plan.JoinNode.Type.RIGHT;
 import static com.facebook.presto.sql.planner.plan.Patterns.join;
+import static java.util.Objects.requireNonNull;
 
 public class DetermineJoinDistributionType
```

Comment:
Do we need to have a similar one for the `SemiJoin`?

Reply:
Hm... we should. But why now?

Reply:
Yeah, we can introduce it as a separate PR. Just in case you can have it stashed somewhere, we can include it here.

Reply:
I don't :(